token-client/README.md

```diff
@@ -1,239 +1,306 @@
+            properties.put("sap.security.services.xsuaa.key", "-----BEGIN RSA PRIVATE KEY ... END RSA PRIVATE KEY-----"); // when using spring-security
           
-          application.setDefaultProperties(properties);
-          application.run(args);
+            application.setDefaultProperties(properties);
+            application.run(args);
           
-      }
-  }
+        }
+    }
+   ```
+
+  :information_source: For **testing purposes only** `key` can be overwritten in `application.yml` properties file.
+   ```yaml
+      # spring-xsuaa
+      xsuaa:
+        key: -----BEGIN RSA PRIVATE KEY ... END RSA PRIVATE KEY-----
+      # spring-security
+      sap.security.services.xsuaa:
+        key: -----BEGIN RSA PRIVATE KEY ... END RSA PRIVATE KEY-----
     ```
-  
-- For **testing purposes only** `key` can be overwritten in `application.yml` properties file.
-    ```yaml
-    # spring-xsuaa
-    xsuaa:
-      key: -----BEGIN RSA PRIVATE KEY-----YOUR PRIVATE KEY-----END RSA PRIVATE KEY-----
-    # spring-security
-    sap.security.services.xsuaa:
-      key: -----BEGIN RSA PRIVATE KEY-----YOUR PRIVATE KEY-----END RSA PRIVATE KEY-----
-  ```
-:exclamation: **DO NOT** disclose your key or secret in publicly available places e.g. repository in github.com
-
-## <a id="OAuth2ServiceConfiguration"></a>Setup OAuth2ServiceConfiguration
-`OAuth2ServiceConfiguration` holds the information from the respective XSUAA service binding. When using `spring-xsuaa` client library this is also given with `XsuaaServiceConfiguration`.
-
-#### Load from Environment
+    :exclamation: **DO NOT** disclose your key or secret in publicly available places e.g. repository in GitHub.com
+
+:bulb: Note that if you are only using the `token-client` library without the [java-security](../java-security/README.md) or [spring-security](../spring-security/README.md), you will need to define the `env` dependency in your pom.xml:
+```xml
+<dependency>
+    <groupId>com.sap.cloud.security</groupId>
+    <artifactId>env</artifactId>
+</dependency>
 ```
-OAuth2ServiceConfiguration config = Environments.getCurrent().getXsuaaConfiguration();
-``` 
-> Note: By default `Environments` auto-detects the environment: Cloud Foundry or Kubernetes. 
 
-#### Instantiate a custom one
-If you need to fetch a token using the uaa service configuration of your job scheduler, you can't use the config loaded from service binding. In that case you may leverage ``OAuth2ServiceConfigurationBuilder`` provided with ``env`` client library. 
+### HttpClientFactory
+`HttpClientFactory` creates an HTTP client that will make the requests to the corresponding Identity service.
+
+#### Autoconfigured tokenFlowHttpClient
+When using `spring-security` client library, a readily configured `CloseableHttpClient` is accessible via `tokenFlowHttpClient` Bean that uses the `HttpClientFactory` internally to set up the HTTP Client for token flows.
```

Comment:
Is spring-security enough or does one also need the auto configuration provided by resourceserver-security-spring-boot-starter for the bean to be available?

Reply:
yeah I guess it could be defined more precisely that boot starter is required otherwise autoconfig doesn't work even thou they are defined in spring-security